toreman

```diff
@@ -1,6 +1,15 @@
 #!/usr/bin/env sh
 
-grep --invert-match '^#' < Procfile |
+while getopts ":f:procfile:" option; do
+  case "${option}" in
+    f) PROCFILE=${OPTARG};;
+    procfile) PROCFILE=${OPTARG};;
+  esac
+done
+
+PROCFILE=${PROCFILE:-Procfile}
+
+grep --invert-match '^#' < $PROCFILE |
```

Comment:
I must admit I've made a huge design mistake by giving in and letting to accept the layout as the first command-line argument (`select-layout ${1:-tiled}`). This prevents any further extensibility attempts without breaking the one-liner approach. It should have been an environment variable, not a positional command-line argument.

To meet the design goal of keeping this script one-line'ish, to keep the ability to set the layout, and to be able to choose the procfile to be used would you agree to do the following:
 - use the `PROCFILE` environment variable, and default to `Procfile`
 - remove the usage of `$1` for setting the layout
 - use `LAYOUT` environment variable for setting the layout, and default to `tiled`
 - update the README to reflect those changes
 - introduce a CHANGELOG.adoc to tell about those changes and mentioning yourself there

I haven't thought of versioning before, and as since this is a breaking change, we would have to change the major version or keep pretending that we were at 0.9 before, and 1.0 now.

Would you like to make such changes?

Reply:
Oh that's fine too! Never bothered to change the tiling. Will the command look like this?

PROCFILE=Procfile.dev.remote toreman

Reply:
> Never bothered to change the tiling.

Me neither 😆 

> Will the command look like this? `PROCFILE=Procfile.dev.remote toreman`

Exactly, that was the idea. Does it fit your needs?